scripts/common/rook.sh

```diff
@@ -368,6 +368,83 @@ function rook_operator_ready() {
+    fi
+
+    log "Awaiting Rook Ceph health ..."
+    if ! $DIR/bin/kurl rook wait-for-health 600 ; then
+        logWarn "Rook Ceph is unhealthy"
+
+        output=$(kubectl -n rook-ceph exec deploy/rook-ceph-tools -- ceph status)
+
+        echo ""
+        echo output
+        echo ""
+
+        if [[ $output == *"Module 'dashboard'"* ]] || [[ $output == *"Module 'prometheus'"* ]]; then
+            echo "Disabling Ceph manager modules in order to get Ceph healthy again."
+            kubectl -n rook-ceph exec deployment/rook-ceph-tools -- ceph mgr module disable prometheus || true
+            kubectl -n rook-ceph exec deployment/rook-ceph-tools -- ceph mgr module disable dashboard || true
+        fi
+
+        log "Verify Rook Ceph health after try to fix"
+        if ! $DIR/bin/kurl rook wait-for-health 600; then
+            kubectl -n rook-ceph exec deploy/rook-ceph-tools -- ceph status
+            logFail "Rook Ceph is unhealthy"
+            return 1
+        fi
+        return 1
+    fi
+
+    log "Checking Rook Ceph versions and replicas"
+    kubectl -n rook-ceph get deployment -l rook_cluster=rook-ceph -o jsonpath='{range .items[*]}{.metadata.name}{"  \treq/upd/avl: "}{.spec.replicas}{"/"}{.status.updatedReplicas}{"/"}{.status.readyReplicas}{"  \trook-version="}{.metadata.labels.rook-version}{"\n"}{end}'
+    local rook_versions=
+    rook_versions="$(kubectl -n rook-ceph get deployment -l rook_cluster=rook-ceph -o jsonpath='{range .items[*]}{"rook-version="}{.metadata.labels.rook-version}{"\n"}{end}' | sort | uniq)"
+    if [ -n "${rook_versions}" ] && [ "$(echo "${rook_versions}" | wc -l)" -gt "1" ]; then
+        logFail "Multiple Rook versions detected"
+        logFail "${rook_versions}"
+        return 1
+    fi
+
+    log "Checking Ceph versions and replicas"
+    kubectl -n rook-ceph get deployment -l rook_cluster=rook-ceph -o jsonpath='{range .items[*]}{.metadata.name}{"  \treq/upd/avl: "}{.spec.replicas}{"/"}{.status.updatedReplicas}{"/"}{.status.readyReplicas}{"  \tceph-version="}{.metadata.labels.ceph-version}{"\n"}{end}'
+    local ceph_versions_found=
+    ceph_versions_found="$(kubectl -n rook-ceph get deployment -l rook_cluster=rook-ceph -o jsonpath='{range .items[*]}{"ceph-version="}{.metadata.labels.ceph-version}{"\n"}{end}' | sort | uniq)"
+    if [ -n "${ceph_versions_found}" ] && [ "$(echo "${ceph_versions_found}" | wc -l)" -gt "1" ]; then
+        # It is required because an Rook Ceph bug which was sorted out with the release 1.4.8
+        # More info: https://github.com/rook/rook/pull/6610
+        if [ "$(echo "${ceph_versions_found}" | wc -l)" == "2" ] && [ "$(echo "${ceph_versions_found}" | grep "0.0.0-0")" ]; then
+            log "Found two ceph versions but one of them is 0.0.0-0 which will be ignored"
+            echo "${ceph_versions_found}"
+        else
+            logFail "Multiple Ceph versions detected"
+            logFail "${ceph_versions_found}"
+            return 1
+        fi
+    fi
```

Comment:
These entire code looks similar to code in `rook_is_healthy_to_upgrade()`. Wouldn't it be better to DRY it by extracting in into some other function

Reply:
I do not think so because that could be risk we introduce an issue.
So, think in the risk vs maintainability I think would be better for now we just keep duplicated and explained why.